opencog/nlp/relex2logic/SV.scm

```diff
@@ -0,0 +1,57 @@
+(define SV
+	(BindLink
+		(ListLink
+			(TypedVariableLink
+				(VariableNode "$a-parse")
+				(TypeNode "ParseNode")
+			)
+			(TypedVariableLink
+				(VariableNode "$subj")
+				(TypeNode "WordInstanceNode")
+			)
+			(TypedVariableLink
+				(VariableNode "$verb")
+				(TypeNode "WordInstanceNode")
+			)
+		)
+		(ImplicationLink
+			(AndLink
+				(WordInstanceLink
+					(VariableNode "$subj")
+					(VariableNode "$a-parse")
+				)
+ 				(WordInstanceLink
+					(VariableNode "$verb")
+					(VariableNode "$a-parse")
+				)
+				(EvaluationLink
+					(DefinedLinguisticRelationshipNode "_subj")
+					(ListLink
+						(VariableNode "$verb")
+						(VariableNode "$subj")
+ 					)
+				)
+			)
+		(ExecutionOutputLink
+			(GroundedSchemaNode "scm: pre-sv-rule")
+			(ListLink
+				(VariableNode "$subj")
+				(VariableNode "$verb")
+			)
+		)
+	)
+))
+
+
+(InheritanceLink (stv 1 .99) (ConceptNode "SV-Rule") (ConceptNode "Rule"))
+
+(ReferenceLink (stv 1 .99) (ConceptNode "SV-Rule") SV)
+
+; This is function is not needed. It is added so as not to break the existing
+; r2l pipeline.
```

Comment:
I don't know; you wrote that!  I copied it into all the rules since i
wasn't sure what you had in mind in terms of not needing the not needed
statement.

On Thu, Apr 30, 2015 at 2:17 PM, Linas Vepštas <notifications@github.com>
wrote:

> In opencog/nlp/relex2logic/SV.scm
> <https://github.com/opencog/opencog/pull/1575#discussion_r29405857>:
>
> > +			(GroundedSchemaNode "scm: pre-sv-rule")
> > +			(ListLink
> > +				(VariableNode "$subj")
> > +				(VariableNode "$verb")
> > +			)
> > +		)
> > +	)
> > +))
> > +
> > +
> > +(InheritanceLink (stv 1 .99) (ConceptNode "SV-Rule") (ConceptNode "Rule"))
> > +
> > +(ReferenceLink (stv 1 .99) (ConceptNode "SV-Rule") SV)
> > +
> > +; This is function is not needed. It is added so as not to break the existing
> > +; r2l pipeline.
>
> In what sense is this "function not needed?" I looked at SV-rule, and it
> seems to do all sorts of complicated stuff, so it does seem to be needed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/opencog/opencog/pull/1575/files#r29405857>.
>